### PR TITLE
feat: Add methods to {Trigger/Job}Collection

### PR DESCRIPTION
**Pull request by ptbrowne**
- Launch a konnector
- Update a konnector
- Wait for the completion of a job

### CODE REVIEW

File: packages/cozy-stack-client/src/KonnectorCollection.js [hunk 1]
```diff
@@ -17,8 +20,44 @@ class KonnectorCollection extends AppCollection {
     throw new Error('destroy() method is not available for konnectors')
   }
 
-  async update() {
-    throw new Error('update() method is not available for konnectors')
+  async launch(slug) {
```
Comment (doubleface) [inline packages/cozy-stack-client/src/KonnectorCollection.js]: An optional accountId (like the cli) or triggerId parameter would be useful when there are more than 1 trigger for the given slug

Comment (ptbrowne) [reply packages/cozy-stack-client/src/KonnectorCollection.js]: Done, thanks. options.accountId can be passed to KonnectorCollection::launch. For a triggerId, the TriggerCollection::launch can be used.

File: packages/cozy-stack-client/src/JobCollection.js [hunk 2]
```diff
@@ -32,11 +45,42 @@ class JobCollection {
...
+   *
+   * `options.until` can be used to tweak when to stop waiting. It will be
+   * given the current job state. If true is returned, the awaiting is
+   * stopped.
+   */
+  async waitFor(id, options = {}) {
+    options.until = options.until || hasJobFinished
+    options.delay = options.delay === undefined ? 5 * 1000 : options.delay
+    options.timeout =
+      options.timeout === undefined ? 60 * 5 * 1000 : options.timeout
```
Comment (Crash--) [inline packages/cozy-stack-client/src/JobCollection.js]: why not defining the default value in the declaration ? I find it clearer to read and understand 

```js
async waitFor(id, options = {
timeout: 60 * 5 * 1000,
delay: 5 * 1000
}) {
```

File: packages/cozy-stack-client/src/KonnectorCollection.js [hunk 3]
```diff
@@ -17,8 +24,73 @@ class KonnectorCollection extends AppCollection {
     throw new Error('destroy() method is not available for konnectors')
   }
 
-  async update() {
-    throw new Error('update() method is not available for konnectors')
+  async findTriggers(slug) {
```
Comment (Crash--) [inline packages/cozy-stack-client/src/KonnectorCollection.js]: - jsdoc 
- findTriggersBySlug(slug) ?

File: packages/cozy-stack-client/src/KonnectorCollection.js [hunk 4]
```diff
@@ -17,8 +24,73 @@ class KonnectorCollection extends AppCollection {
...
+    const triggerCol = new TriggerCollection(this.stackClient)
+    const { data: rawTriggers } = await triggerCol.all()
```
Comment (Crash--) [inline packages/cozy-stack-client/src/KonnectorCollection.js]: all() is paginated no? So we are trying to `find triggers by slug in the last 100 triggers` ?

File: packages/cozy-stack-client/src/KonnectorCollection.js [hunk 5]
```diff
@@ -17,8 +24,73 @@ class KonnectorCollection extends AppCollection {
...
+    }
+  }
+
+  /**
+   * Updates a konnector
+   *
+   * @param  {String} slug
+   * @param  {Object} options
+   * @param  {Object} options.source - Specify the source (ex: registry://slug/stable)
+   * @param  {Object} options.sync - Wait for konnector to be updated, otherwise the job
```
Comment (Crash--) [inline packages/cozy-stack-client/src/KonnectorCollection.js]: I don't find `sync` is very clear. Is it a boolean?

File: packages/cozy-stack-client/src/JobCollection.js (hunk 2 above)
Comment (ptbrowne) [reply packages/cozy-stack-client/src/JobCollection.js]: ca n'a pas le meme comportement si tu passes un objet partiel d'options.

avec ton implementation, si tu fais:

```
waitFor(id, { delay: 3 * 1000 })
```

tu perds le timeout par défaut.

File: packages/cozy-stack-client/src/KonnectorCollection.js (hunk 5 above)
Comment (ptbrowne) [reply packages/cozy-stack-client/src/KonnectorCollection.js]: Yes, I fixed the type of the param, do you have any idea to make it clearer ?

File: packages/cozy-stack-client/src/KonnectorCollection.js (hunk 4 above)
Comment (ptbrowne) [reply packages/cozy-stack-client/src/KonnectorCollection.js]: Thx, I added `{ limit: null }`

File: packages/cozy-stack-client/src/KonnectorCollection.js (hunk 3 above)
Comment (ptbrowne) [reply packages/cozy-stack-client/src/KonnectorCollection.js]: Done, thanks

File: packages/cozy-stack-client/src/JobCollection.js (hunk 2 above)
Comment (Crash--) [reply packages/cozy-stack-client/src/JobCollection.js]: Capture d’écran 2019-10-04 à 14 36 24

🤔 . J'ai mal compris un truc ?

Comment (ptbrowne) [reply packages/cozy-stack-client/src/JobCollection.js]:
```
> fn = (opts = { a: 1, b: 2}) => { console.log(opts.a, opts.b) }
[Function: fn]
> fn()
1 2
undefined
> fn({a:0})
0 undefined // 😱 no b !
```

Comment (ptbrowne) [reply packages/cozy-stack-client/src/JobCollection.js]: Fixed by using the destructing method ({ arg = defaultValue }). Thanks

**Pull request merged**